File: leetcode/editor/cn/226_翻转二叉树_test.py
import unittest
from typing import Optional

from model.tree_node import TreeNode


class TestSolution(unittest.TestCase):

    def setUp(self) -> None:
        self.solution = Solution()

    def test_givenNormal_when_thenSuccess(self):
        self.assertTrue(True, "ShouldBeTrue")


# leetcode submit region begin(Prohibit modification and deletion)
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def invertTree(self, root: Optional[TreeNode]) -> Optional[TreeNode]:
        if root:
            left = self.invertTree(root.left)
            right = self.invertTree(root.right)
            root.left, root.right = right, left
        return root


# leetcode submit region end(Prohibit modification and deletion)


if __name__ == '__main__':
    unittest.main()

File: leetcode/editor/cn/226_翻转二叉树_test.go
package leetcode

import (
	"github.com/stretchr/testify/assert"
	"testing"
)

func Test_givenNormal_when226_thenSuccess(t *testing.T) {
	tree := decodeTreeNode([]any{4, 2, 7, 1, 3, 6, 9})
	invertedTree := invertTree(tree)
	assert.Equal(t, []any{4, 7, 2, 9, 6, 3, 1}, encodeTreeNode(invertedTree))
}

//leetcode submit region begin(Prohibit modification and deletion)
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func invertTree(root *TreeNode) *TreeNode {
	if root != nil {
		// switch left and right
		root.Left, root.Right = invertTree(root.Right), invertTree(root.Left)
	}
	return root
}

//leetcode submit region end(Prohibit modification and deletion)
